File: 20180710/Find All Numbers Disappeared In An Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList();
        int length = nums.length;
        if (length <= 1){
            return res;
        }
        boolean[] check = new boolean[length];
        for(int i = 0; i < length; i++){
           check[nums[i] - 1] = true;
        }
        for(int i = 0; i < length; i++){
           if(!check[i]){
               res.add(i+1);
           }
        }
        return res;
    }
}

File: 20180710/Find All Numbers Disappeared In An Array.py
class Solution(object):
    def exchange(self,a,b):
        return b,a
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # I am so stupid to make this so complicated..
        # for i in range(len(nums)):
        #     while nums[i] != i + 1 and nums[nums[i] - 1] != nums[i]:
        #         nums[nums[i] - 1] , nums[i]= self.exchange(nums[nums[i] - 1],nums[i])
        # return [i+1 for i in range(len(nums)) if i + 1 != nums[i]]
        length = len(nums)
        nums = set(nums)
        res = []
        for i in range(length):
            if i + 1 not in nums:
                res.append(i + 1)
        return res
        